inst/bs3compat/_shiny_input.scss

```
@@ -46,3 +46,27 @@ $form-group-margin-bottom: 1rem !default;
     border-radius: 0;
   }
 }
+
+.shiny-input-checkboxgroup .checkbox-inline,
+.shiny-input-radiogroup .radio-inline {
+  cursor: pointer;
+}
+
+// Fix the position of checkbox and radio group labels to match other inputs
+// labels, see https://github.com/rstudio/shiny/blob/6fc06281/inst/www/shared/shiny_scss/shiny.scss#L344-L356
+@if $bootstrap-version == 4 {
+  .shiny-input-checkboxgroup,
+  .shiny-input-radiogroup {
+    label ~ .shiny-options-group {
+      // Margin is in rem/em and border is px and Sass math requires same units
+      margin-top: calc(-1 * #{$form-check-input-margin-y} - #{$input-border-width});
+    }
+  }
+} @else if $bootstrap-version == 5 {
+  .shiny-input-checkboxgroup,
+  .shiny-input-radiogroup {
+    label ~ .shiny-options-group {
+      margin-top: calc(#{-($line-height-base - $form-check-input-width) * .5} - #{$input-btn-border-width});
+    }
+  }
+}
```

I appreciate you looking into the BS4 codepaths, but I don't think it's critical that we improve BS4? In general, I think we should be taking an approach of only taking on major security / bug fixes for BS4 

```suggestion
// Fix the position of checkbox and radio group labels to match other inputs
// labels, see https://github.com/rstudio/shiny/blob/6fc06281/inst/www/shared/shiny_scss/shiny.scss#L344-L356
.shiny-input-checkboxgroup, .shiny-input-radiogroup {
  label ~ .shiny-options-group {
    margin-top: calc(#{-($line-height-base - $form-check-input-width) * .5} - #{$input-btn-border-width});
  }
}
```

---

```suggestion
// Fix the position of checkbox and radio group labels to match other inputs
// labels, see https://github.com/rstudio/shiny/blob/6fc06281/inst/www/shared/shiny_scss/shiny.scss#L344-L356
@if $bootstrap-version == 5 {
  .shiny-input-checkboxgroup,
  .shiny-input-radiogroup {
    label ~ .shiny-options-group {
      margin-top: calc(#{-($line-height-base - $form-check-input-width) * .5} - #{$input-btn-border-width});
    }
  }
}
```

Sounds good but we still need the BS5 check since some of those variables don't exist in BS4.

---

Done in 89ff4c50d54a841eaffbc768c163944a3b1b058e